rust/blockstore/src/arrow/block/value/data_record_value.rs

```diff
@@ -45,6 +67,116 @@ impl ArrowWriteableValue for &DataRecord<'_> {
     fn get_delta_builder() -> BlockStorage {
         BlockStorage::DataRecord(DataRecordStorage::new())
     }
+
+    fn get_arrow_builder(size_tracker: Self::SizeTracker) -> Self::ArrowBuilder {
+        ValueBuilderWrapper {
+            id_builder: StringBuilder::with_capacity(
+                size_tracker.get_num_items(),
+                size_tracker.get_id_size(),
```

Comment:
confirming 

`get_id_size()` will return size of _all_ strings summed

Reply:
yep 👍